pennylane/gradients/hadamard_gradient.py

```diff
@@ -329,7 +329,8 @@ def _expval_hadamard_grad(tape, argnum, aux_wire):
                     obs_new = [qml.Z(i) for i in m.wires]
 
                 obs_new.append(qml.Y(aux_wire))
-                obs_new = qml.operation.Tensor(*obs_new)
+                obs_type = qml.prod if qml.operation.active_new_opmath() else qml.operation.Tensor
+                obs_new = obs_type(*obs_new)
```

Comment:
Why not just use the `@` operator?

Reply:
I believe `obs_new` is of varying length, so I'm not sure how to do that nicely

Reply:
Looks good, saw also christina do that somewhere else, neat solution 👍